yt_dlp/extractor/dlf.py

```diff
@@ -0,0 +1,209 @@
+                    'uploader': 'Deutschlandfunk',
+                    'series': 'Sport am Samstag',
+                    'channel': 'deutschlandfunk'
+                }
+            }, {
+                'info_dict': {
+                    'id': '47e1a096',
+                    'title': r're:Rückkehr Russlands im Fechten [-/] "Fassungslos, dass es einfach so passiert ist"',
+                    'url': r're:https://(?:download\.deutschlandfunk|ondemand-mp3\.dradio)\.de/file/dradio/2023/03/11/weltfechtverband_laesst_russen_und_belarussen_wieder_zu_dlf_20230311_1912_47e1a096\.mp3',
+                    'ext': 'mp3',
+                    'duration': 602,
+                    'thumbnail': 'https://assets.deutschlandfunk.de/da4c494a-21cc-48b4-9cc7-40e09fd442c2/512x512.jpg?t=1678562155770',
+                    'uploader': 'Deutschlandfunk',
+                    'series': 'Sport am Samstag',
+                    'channel': 'deutschlandfunk'
+                }
+            }, {
+                'info_dict': {
+                    'id': '5e55e8c9',
+                    'title': r're:Wiederzulassung von Russland und Belarus [-/] "Herumlavieren" des Fechter-Bundes sorgt für Unverständnis',
+                    'url': r're:https://(?:download\.deutschlandfunk|ondemand-mp3\.dradio)\.de/file/dradio/2023/03/18/fechten_unverstaendnis_und_wut_nach_wiederzulassung_dlf_20230318_1930_5e55e8c9\.mp3',
+                    'ext': 'mp3',
+                    'duration': 187,
+                    'thumbnail': 'https://assets.deutschlandfunk.de/a595989d-1ed1-4a2e-8370-b64d7f11d757/512x512.jpg?t=1679173825412',
+                    'uploader': 'Deutschlandfunk',
+                    'series': 'Sport am Samstag',
+                    'channel': 'deutschlandfunk'
+                }
+            }],
+            'params': {'skip_download': 'm3u8'}
+        },
+        # Podcast feed with tag buttons
+        {
+            'url': 'https://www.deutschlandfunk.de/corso-100.html',
+            'info_dict': {
+                'id': False,
+                'title': r're:Kunst & Pop [-/] Corso',
+                'description': 'md5:ba0f80303683591bff4b7459077bef58'
+            },
+            'playlist_mincount': 34,
+            'params': {'skip_download': True}
+        },
+        # Podcast feed with no description
+        {
+            'url': 'https://www.deutschlandfunk.de/podcast-tolle-idee-100.html',
+            'info_dict': {
+                'id': False,
+                'title': 'Wissenschaftspodcast - Tolle Idee! - Was wurde daraus?',
+                'description': None
+            },
+            'playlist_mincount': 11,
+            'params': {'skip_download': True}
+        },
+    ]
+
+    def _real_extract(self, url):
+        # Multi-feed pages do not embed an ID within the URL
+        webpage = self._download_webpage(url, None)
+
+        return {
+            '_type': 'playlist',
+            'id': False,
```

Comment:
Is the URL slug unique? Can't we match that for the playlist `id`?

Reply:
Oh, I didn't even consider using the entire URL slug. My mind was fixated on the trailing digits in the slug, which is why I jumped to that conclusion.

I updated the regex to correlate the slug with the `id`:

```python
DLFBaseIE._VALID_URL_BASE + r'(?P<id>(?:[a-z]+-)+\d+)\.html'
```

Should I leave the nested `?:` within `id`'s capture group?

Reply:
```py
DLFBaseIE._VALID_URL_BASE + r'(?P<id>[\w-]+-\d+)\.html'
```
is the preferred way

Reply:
I hadn't realized how similar the DFL and DFLCorpus URL patterns were when I suggested that, my bad

do DFL single media URLs always end with `dlf-[\da-f]{6}-100\.html` e.g. `dlf-03a3eb19-100.html` ?

Reply:
> I hadn't realized how similar the DFL and DFLCorpus URL patterns were when I suggested that, my bad

No worries!

> do DLF single media URLs always end with -dlf-[\da-f]{8}-100\.html e.g. -dlf-03a3eb19-100.html

Yeah, they do. I checked out a bunch of sites while creating the extractor, and that seems to be the case.